platform-sdk/docs/proposals/state-refactor/state-refactor.md

```diff
@@ -0,0 +1,211 @@
+# State Refactor for State Proofs
```

Comment:
This document is missing test related considerations, which are meant to be included here.

Reply:
Added test related considerations